xla/service/cpu/onednn_memory_util.cc

```diff
@@ -74,14 +74,22 @@ MemrefInfoHandler CreateMemrefInfoFromLiteral(const Literal* literal) {
 StackAlloca GetAllocaAndEmitMemrefInfo(llvm::IRBuilder<>& builder,
                                        const llvm_ir::IrArray& ir_array) {
   const Shape& shape = ir_array.GetShape();
-  int64_t rank = shape.rank();
-  absl::Span<const int64_t> dims = shape.dimensions();
+  // oneDNN handles scalar as a vector of size 1.
+  int64_t rank = shape.rank() == 0 ? 1 : shape.rank();
+  std::vector<int64_t> scalar_shape(1, 1);
+  absl::Span<const int64_t> dims =
+      shape.dimensions().size() == 0 ? scalar_shape : shape.dimensions();
 
   std::vector<int64_t> strides(rank);
-  int64_t stride = 1;
-  for (int i : shape.layout().minor_to_major()) {
-    strides.at(i) = stride;
-    stride *= dims.at(i);
+  if (shape.dimensions().size() == 0) {
+    // Scalar case.
+    strides[0] = 1;
+  } else {
+    int64_t stride = 1;
+    for (int i : shape.layout().minor_to_major()) {
+      strides.at(i) = stride;
+      stride *= dims.at(i);
+    }
```

Comment:
Could you please refactor this code to a utility function and call it from both `GetAllocaAndEmitMemrefInfo` and `ShapeToMemDesc`?

Reply:
Done.